### PR TITLE
Added option to force local login and gracefully continue after failed vCards

### PR DESCRIPTION
**Pull request by ralf1070**
- warn about failed downloads of vCards but do not abort
- add option to force login on local fritzbox account even when FQDN of fritzbox is not fritz.box

### CODE REVIEW

File: carddav2fb.php [hunk 1]
```diff
@@ -504,7 +508,8 @@ public function upload_to_fb() {
     }
     catch (Exception $e)
     {
-      print "  ERROR: " . $e->getMessage() . PHP_EOL;     // show the error message in anything failed
+//      print "  ERROR: " . $e->getMessage() . PHP_EOL;     // show the error message in anything failed
+      print "  ERROR: " . $e . PHP_EOL;     // show the error message in anything failed
```
Comment (jens-maus) [inline carddav2fb.php]: can you please explain why you replace $e->getMessage() here with plainly $e? I think it still should use $e->getMessage().

File: carddav2fb.php [hunk 2]
```diff
@@ -44,6 +44,7 @@
 $config['tmp_dir'] = sys_get_temp_dir();
 $config['fritzbox_ip'] = 'fritz.box';
 $config['fritzbox_ip_ftp'] = 'fritz.box';
+$config['fritzbox_force_local_config'] = false;
```
Comment (jens-maus) [inline carddav2fb.php]: Please rename this option to something like ```$config['fritzbox_force_local_login']``` as this better explains what this option is actually doing.

File: carddav2fb.php (hunk 1 above)
Comment (ralf1070) [inline carddav2fb.php]: > Am 18.12.2015 um 14:59 schrieb Jens Maus <notifications@github.com>:
> 
> In carddav2fb.php:
> 
> > @@ -504,7 +508,8 @@ public function upload_to_fb() {
> >      }
> >      catch (Exception $e)
> >      {
> > -      print "  ERROR: " . $e->getMessage() . PHP_EOL;     // show the error message in anything failed
> > +//      print "  ERROR: " . $e->getMessage() . PHP_EOL;     // show the error message in anything failed
> > +      print "  ERROR: " . $e . PHP_EOL;     // show the error message in anything failed
> 
> can you please explain why you replace $e->getMessage() here with plainly $e? I think it still should use $e->getMessage().

Ah sorry - debug artefact. I needed to know where the error came from. $e-getMessage() is better here.

File: carddav2fb.php (hunk 2 above)
Comment (ralf1070) [inline carddav2fb.php]: > Am 18.12.2015 um 15:02 schrieb Jens Maus <notifications@github.com>:
> 
> In carddav2fb.php:
> 
> > @@ -44,6 +44,7 @@
> >  $config['tmp_dir'] = sys_get_temp_dir();
> >  $config['fritzbox_ip'] = 'fritz.box';
> >  $config['fritzbox_ip_ftp'] = 'fritz.box';
> > +$config['fritzbox_force_local_config'] = false;
> 
> Please rename this option to something like $config['fritzbox_force_local_login'] as this better explains what this option is actually doing.

Done.

**Comment by ralf1070** [comment]
Better this way?

File: lib/fritzbox_api_php/fritzbox_api.class.php [hunk 3]
```diff
@@ -63,7 +63,7 @@ class fritzbox_api {
     * 
     * @access public
     */
-  public function __construct($password = false,$user_name = false,$fritzbox_ip = 'fritz.box')
+  public function __construct($password = false,$user_name = false,$fritzbox_ip = 'fritz.box',$force_local_config = false)
```
Comment (jens-maus) [inline lib/fritzbox_api_php/fritzbox_api.class.php]: Then please rename this also to ```$force_local_login```

**Comment by jens-maus** [comment]
Almost done, yes ;)

**Comment by ralf1070** [comment]
So we do.

**Comment by jens-maus** [comment]
Done! Thanks for this nice pull request!

**Comment by ralf1070** [comment]
Thank you too.

**Pull request merged**